src/compiler/builder.ts

```diff
@@ -853,8 +853,13 @@ namespace ts {
         const { optionsNameMap } = getOptionsNameMap();
 
         for (const name of getOwnKeys(options).sort(compareStringsCaseSensitive)) {
-            const optionInfo = optionsNameMap.get(name.toLowerCase());
-            if (optionInfo?.affectsEmit || optionInfo?.affectsSemanticDiagnostics || name === "skipLibCheck" || name === "skipDefaultLibCheck") {
+            const optionKey = name.toLowerCase();
+            const optionInfo = optionsNameMap.get(optionKey);
+            if (optionInfo?.affectsEmit || optionInfo?.affectsSemanticDiagnostics ||
+                // Since we should never compare strict directly between options but individual flag, but it does change value of that flag so store it

```

Comment:
Sorry, I don't understand this comment.

Reply:
I think it might be saying that we actually make decisions based on the collection of flags that make up `strict` and not based on `strict` itself, but then I don't understand why would need to store `strict`, rather than those component flags.

Reply:
Oh, maybe deserializing `strict` results in the component flags being set appropriately and then the comparison that looks at those flags and not `strict` behaves correctly?  In which case, maybe something like

> We need to store `strict`, even though it won't be examined directly, so that the flags it controls (e.g. `strictNullChecks`) will be retrieved correctly from the buildinfo

Reply:
```suggestion
                // We need to store `strict`, even though it won't be examined directly, so that the
                // flags it controls (e.g. `strictNullChecks`) will be retrieved correctly from the buildinfo